wordpress-mobile/WordPress-iOS
# [Gutenberg] Audio upload processor

`gutenberg` PR: https://github.com/WordPress/gutenberg/pull/27401
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2854

This PR implements the Audio upload processor to be used together with the Audio block.

This PR also improves the file type filet for the on-device file picker: Now only file extensions allowed by the blog can be picked. Previously any video/image extensions could be picked for the Audio/Video blocks. This is also now used on the Audio block to avoid Apple-only audio filetypes (common on audio files recorded in iPhones).

**To test:**
Test running metro server on https://github.com/wordpress-mobile/gutenberg-mobile/pull/2854

- On a new post, add an Audio block.
- Add to it a relatively large Audio file.
- While the upload is happening, close the editor saving as a draft.
  - The post card will show an upload in progress.
- Wait for the upload to finish and the post to be saved remotely.
- Open the saved post.
- Switch to HTML mode.
  - Check that the Audio block id has a positive number.
  - Check that the `src` URL is a remote (non-local) URL.

cc @jd-alexander 

PR submission checklist:
